server/jetstream_test.go

```diff
@@ -21612,3 +21612,116 @@ func TestJetStreamLimitsToInterestPolicy(t *testing.T) {
+	}
+
+	nc := clientConnectToServer(t, s)
+	defer nc.Close()
+
+	js, err := nc.JetStream()
+	if err != nil {
+		t.Fatalf("Failed to connect to JetStream: %v", err)
+	}
+
+	t.Run("InheritDefaultsFromStream", func(t *testing.T) {
+		ci, err := js.AddConsumer("test", &nats.ConsumerConfig{
+			Name: "InheritDefaultsFromStream",
+		})
+		require_NoError(t, err)
+
+		switch {
+		case ci.Config.InactiveThreshold != time.Second:
+			t.Fatalf("InactiveThreshold should be 1s, got %s", ci.Config.InactiveThreshold)
+		case ci.Config.MaxAckPending != 15:
+			t.Fatalf("MaxAckPending should be 15, got %d", ci.Config.MaxAckPending)
+		}
+	})
+
+	t.Run("CreateConsumerErrorOnExceedMaxAckPending", func(t *testing.T) {
+		_, err := js.AddConsumer("test", &nats.ConsumerConfig{
+			Name:          "CreateConsumerErrorOnExceedMaxAckPending",
+			MaxAckPending: 30,
+		})
+		switch e := err.(type) {
+		case *nats.APIError:
+			if ErrorIdentifier(e.ErrorCode) != JSConsumerMaxPendingAckExcessErrF {
+				t.Fatalf("invalid error code, got %d, wanted %d", e.ErrorCode, JSConsumerMaxPendingAckExcessErrF)
+			}
+		default:
+			t.Fatalf("should have returned API error, got %T", e)
+		}
+	})
+
+	t.Run("CreateConsumerErrorOnExceedInactiveThreshold", func(t *testing.T) {
+		_, err := js.AddConsumer("test", &nats.ConsumerConfig{
+			Name:              "CreateConsumerErrorOnExceedInactiveThreshold",
+			InactiveThreshold: time.Second * 2,
+		})
+		switch e := err.(type) {
+		case *nats.APIError:
+			if ErrorIdentifier(e.ErrorCode) != JSConsumerInactiveThresholdExcess {
+				t.Fatalf("invalid error code, got %d, wanted %d", e.ErrorCode, JSConsumerInactiveThresholdExcess)
+			}
+		default:
+			t.Fatalf("should have returned API error, got %T", e)
+		}
+	})
+
+	t.Run("UpdateStreamErrorOnViolateConsumerMaxAckPending", func(t *testing.T) {
+		_, err := js.AddConsumer("test", &nats.ConsumerConfig{
+			Name:          "UpdateStreamErrorOnViolateConsumerMaxAckPending",
+			MaxAckPending: 15,
+		})
+		require_NoError(t, err)
+
+		stream, err := acc.lookupStream("test")
+		require_NoError(t, err)
+
+		err = stream.update(&StreamConfig{
+			Name:     "test",
+			Subjects: []string{"test.*"},
+			ConsumerLimits: StreamConsumerLimits{
+				MaxAckPending: 10,
+			},
+		})
+		if err == nil {
+			t.Fatalf("stream update should have errored but didn't")
+		}
+	})
+
+	t.Run("UpdateStreamErrorOnViolateConsumerInactiveThreshold", func(t *testing.T) {
+		_, err := js.AddConsumer("test", &nats.ConsumerConfig{
+			Name:              "UpdateStreamErrorOnViolateConsumerInactiveThreshold",
+			InactiveThreshold: time.Second,
+		})
+		require_NoError(t, err)
+
+		stream, err := acc.lookupStream("test")
+		require_NoError(t, err)
+
+		err = stream.update(&StreamConfig{
+			Name:     "test",
+			Subjects: []string{"test.*"},
+			ConsumerLimits: StreamConsumerLimits{
+				InactiveThreshold: time.Second / 2,
+			},
+		})
+		if err == nil {
+			t.Fatalf("stream update should have errored but didn't")
+		}
+	})
```

Comment:
My main concern about this approach, without `force` flag, is that it will generate a lot of frinction.
If someone has a `Stream` with a lot of Consumers, with most setting set to default, **every** update of stream limits will fail, listing all consumers that violate the limits.

Users might be frustrated that now they have to manually update potentially hunders of Consumers before being able to set the limits.

Test below shows that this is the case even for simple default consumer:

```
diff --git a/server/jetstream_cluster_3_test.go b/server/jetstream_cluster_3_test.go
index 2cf55f34..657f816a 100644
--- a/server/jetstream_cluster_3_test.go
+++ b/server/jetstream_cluster_3_test.go
@@ -5232,4 +5232,25 @@ func TestJetStreamClusterConsumerDefaultsFromStream(t *testing.T) {
                        t.Fatalf("stream update should have errored but didn't")
                }
        })
+
+       t.Run("UpdateStreamWithExistingConsumerWithDefaults", func(t *testing.T) {
+
+               // Add a Stream without limits.
+               _, err := js.AddStream(&nats.StreamConfig{
+                       Name: "unlimited",
+               })
+               require_NoError(t, err)
+
+               // Add a default Consumer.
+               _, err = js.AddConsumer("unlimited", &nats.ConsumerConfig{
+                       Name: "UpdateStreamErrorOnViolateConsumerMaxAckPending",
+               })
+               require_NoError(t, err)
+
+               // Set Limits
+               if _, err = streamUpdate(10, 0); err == nil {
+                       t.Fatalf("stream update should have errored but didn't")
+               }
+
+       })
 }
 ```